src/core/CorePool.ts

```diff
@@ -325,12 +409,56 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): Position {
-    // TODO
-    return new Position();
+    let _position: Position = this.getPosition(owner, tickLower, tickUpper);
+
+    let _tick = this.tickManager.getTickAndInitIfAbsent(this._tickCurrent);
```

Comment:
Nit: _ as prefix on properties of a class is not conducted willingly but I have to do that for getter/setter convention in Typescript. So let's reduce using _ as prefix unless it's necessary.

Reply:
Ok, will follow this format